playbooks/roles/prometheus/defaults/main.yml

```diff
@@ -6,11 +6,12 @@ helm_binary_path: /usr/local/bin/helm
 deploy_prometheus: yes
 create_dashboard: yes
 alertmanager_notification_activated: false
-# slack_webhook_url: "https://hooks.slack.com/services/xxxxxxxxxxxxxxx/xxxxxxxxxxxxxxx/xxxxxxxxxxxxxxx" 
+# slack_webhook_url: "https://hooks.slack.com/services/xxxxxxxxxxxxxxx/xxxxxxxxxxxxxxx/xxxxxxxxxxxxxxx"
 prometheus_crds_url:
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
-- https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
+  - https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
+storage_class:  "{% if cloud_provider is defined and cloud_provider == 'aws' %} gp2 {% else %} default {% endif %}"
```

Comment:
> Should we make it default by the following based on the doc?
> ````
> kubectl annotate storageclass gp2 storageclass.kubernetes.io/is-default-class=true
> ````

From the output below and the documentation, it appears that `gp2` StorageClass is already created as the default StorageClass prior to EKS Kubernetes version 1.11 (because it is marked as `(default)`).

````
[centos@bastion-1 ~]$ kubectl get storageclass
NAME PROVISIONER RECLAIMPOLICY VOLUMEBINDINGMODE ALLOWVOLUMEEXPANSION AGE
gp2 (default) kubernetes.io/aws-ebs Delete WaitForFirstConsumer false 29m
````

- https://docs.aws.amazon.com/eks/latest/userguide/storage-classes.html

There are two methods that can be considered here.

One option is to explicitly specify the StorageClass, as this PR does now, which allows us to specify the appropriate StorageClass for each cloud provider, but requires us to manage which StorageClass to use for each cloud provider.

The other option is to not specify a StorageClass and use the default StorageClass. In this case, do not set it unless `storageclass` is  explicitly specified:

```
    storageSpec:
      volumeClaimTemplate:
        spec:
{% if storageclass is defined and storageclass != "" %}
          storageClassName: {{ storage_class }}
{% endif %}
```

The only thing we need to worry about with this method is which StorageClass is specified for default StorageClass in Azure Kubernetes Engine, but according to the documentation, `default` StorageClass is configured to be used as default StorageClass.

> If no StorageClass is specified for a persistent volume, the default StorageClass is used.

- https://docs.microsoft.com/ja-jp/azure/aks/concepts-storage#storage-classes

Since default StorageClass is used, there is no need to manage StorageClass name for each cloud provider, but default StorageClass may be changed without our knowledge.

I suggest choosing the second option, "Use default StorageClass".

Reply:
@superbrothers Thank you for the clarification!
@tei-k Let's take the second option as suggested.

Reply:
@superbrothers Thanks!

I've confirmed that it works without specifying `storageClassName`.
However, Maybe we don't need the ability to specify the storage class externally.

Can we always use the `default` storage class for prometheus?

Reply:
If you are only thinking about internal use, then "always use the default StorageClass" may be fine. However, if you want to be used by users, you should be able to customize it. Also, if something that is currently variable is fixed, it will be a functional degradation. It all depends on user’s needs.

Reply:
@superbrothers Thanks! I know what you mean.

Currently, the storage capacity is also not customizable, so what else should be customized in prometheus?

In this PR, I want to just make sure that EKS can use the default storage class as well as AKS!

Reply:
@tei-k Why don't you make it customizable as suggested above? 
```
    storageSpec:
      volumeClaimTemplate:
        spec:
{% if storageclass is defined and storageclass != "" %}
          storageClassName: {{ storage_class }}
{% endif %}
```

Reply:
@feeblefakie 

Because this PR just for bugfix, customization is out of scope since it is a functional improvement.

I think it would be better to make a list of items that need customization and fix it in other PR.

Reply:
@tei-k OK! I agree.

Reply:
> Currently, the storage capacity is also not customizable, so what else should be customized in prometheus?

I mistakenly assumed that it is still customizable. No additional features will be needed unless requested.